src/p2p/Join/v2/index.ts

```diff
@@ -248,6 +248,14 @@ export function isOnStandbyList(publicKey: string): boolean {
   }
 }
 
+export function isInNewJoinRequests(publicKey: string): boolean {
```

Comment:
This could be simplified

```suggestion
export function isInNewJoinRequests(publicKey: string): boolean {
  return newJoinRequests.some((node) => node.nodeInfo.publicKey === publicKey);
}
```